src/Sulu/Bundle/DocumentManagerBundle/Command/PhpCRCleanerCommand.php

```diff
@@ -0,0 +1,244 @@
+                }
+
+                if (null === $metaData) {
+                    $output->writeln(\sprintf(
+                        '[Skipping] Unable to load metaData for structure with type "page" and "%s"',
+                        $template
+                    ));
+                    continue;
+                }
+
+                $this->migrate('', $contentData, $metaData, 0);
+            }
+
+            $paths = $this->propertyParser->keyIterator($this->getNodesToDeleteAndClear());
+            foreach ($paths as $propertyPath) {
+                try {
+                    $node->getProperty($propertyPath)->remove();
+                    ++$i;
+                } catch (PathNotFoundException) {
+                    continue;
+                }
+            }
+        }
+        $this->session->save();
+
+        $this->output->writeln('Deleted properties: ' . $i);
+
+        return 0;
+    }
+
+    /**
+     * @param 16|32|64|128|256 $verbosity
+     */
+    private function depthDump(string $message, int $depth, int $verbosity = OutputInterface::VERBOSITY_NORMAL): void
+    {
+        $this->output->writeln(\str_repeat('  ', $depth) . $message, $verbosity);
+    }
+
+    /**
+     * @param array<mixed>|null $data
+     */
+    public function migrate(string|int $key, $data, object $metaData, int $depth): void
+    {
+        // If we are at the top of the metadata run we don't want to mess with that as it contains extra properties like seo and other extension data
+        if ($metaData instanceof StructureMetadata) {
+            foreach ($metaData->getProperties() as $subKey => $value) {
+                $this->depthDump('Iterating over: ' . $subKey, $depth, OutputInterface::VERBOSITY_VERY_VERBOSE);
+
+                /** @var array<mixed>|null $subValue */
+                $subValue = $data[$subKey] ?? null;
+                $this->migrate($subKey, $subValue, $value, $depth);
+            }
+
+            return;
+        }
+
+        if ($metaData instanceof BlockMetadata) {
+            if (null === $data) {
+                Assert::eq(0, $depth, 'Empty arrays are only allowed at the root');
+                $this->depthDump('Empty page? ' . $metaData->getName(), $depth, OutputInterface::VERBOSITY_VERY_VERBOSE);
+            } else {
+                $this->handleBlockMetadata($data, $metaData, $depth);
+            }
+        } elseif ($metaData instanceof ComponentMetadata) {
+            if (null === $data) {
+                Assert::eq(0, $depth, 'Empty arrays are only allowed at the root');
+                $this->depthDump('Empty page? ' . $metaData->getName(), $depth, OutputInterface::VERBOSITY_VERY_VERBOSE);
+            } else {
+                $this->handleComponentMetadata($data, $metaData, $depth);
+            }
+        } elseif ($metaData instanceof PropertyMetadata) {
+            if (null === $data) {
+                Assert::eq(0, $depth, 'Empty arrays are only allowed at the root');
```

Comment:
hmm this is also quite strange why is null or empty arrays are only allowed in the root level